### DISCUSSION TITLE
Nginx proxy - Static files

### DISCUSSION THREAD

**Discussion by Scot-Survivor** [plan-player-analytics/Plan#2425] [Q&A] [answered]
So, I am trying to use Nginx as a reverse proxy. However, this proxy is at "https://scot-survivor.space/plan", while it forwards correctly, I can't work out how to forward static files. Currently, my Nginx config is this:
```
server {
    listen         80 default_server;
    listen         [::]:80 default_server;
    location /blackjack/ {
       alias           /var/www/blackjack-app.com/;
       index          index.html;
       autoindex on;
    }
    location /plan {
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_pass http://127.0.0.1:8804;
    }
}
server {
    if ($http_x_referer = "https://scot-survivor.space/plan") {
       return 301 https://$host/plan$request_uri;
    }
    location /blackjack {
       alias           /var/www/blackjack-app.com/;
       index          index.html;
       autoindex on;
    }
    location /plan {
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_pass http://127.0.0.1:8804;
        alias      /var/www/plan-test.com/;
    }

    server_name scot-survivor.space; # managed by Certbot

    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/scot-survivor.space/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/scot-survivor.space/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}server {
    if ($host = scot-survivor.space) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    listen         80 ;
    listen         [::]:80 ;
    server_name scot-survivor.space;
    return 404; # managed by Certbot
}
```
Using burp suite, I saw that the "referer" header may be able to be used in order to check if it's being referred by `<IP>/plan`. However, this still isn't working. 
In plan config, is it possible to change the "web static path" I.E instead of always going to the root of the web address, can I set it so it looks at `/plan`? 
Therefore, each static request will be 
`/plan/css`... 
etc etc

I could ctrl+f the HTML files so that all resource requests go to `/plan`, but that seems a little too hacky. 

Let me know if you need my Plan config too

Thank you for all your help. 
-Scot_Survivor

**Comment by AuroraLS3** [comment, marked as answer]
Please refer to https://github.com/plan-player-analytics/Plan/wiki/External-Webserver-use:-Reverse-proxy#nginx-reverse-proxy and nginx documentation.

Any location should end with `/` if using it as a reverse proxy directory.

If you use subdirectory the proxy_pass needs to end with `/` as well - eg
```nginx
server {
    listen 80;
    server_name scot-survivor.space;

    location /plan/ {
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_pass http://127.0.0.1:8804/;
    }
}
```

----

Your question is confusing me since you want to direct traffic to Plan webserver but also serve static files for Plan even though Plan webserver serves those.

nginx has try_files directive for serving static files https://linuxhint.com/use-nginx-try_files-directive/ and there's a possibility of having nginx try address and then try static files or other way around by defining `@<location>` names for the `try_files` directive https://stackoverflow.com/questions/36510523/nginx-try-files-proxy-pass-and-upstream/36556619#36556619

**Reply by Scot-Survivor** [reply]
Thank you, I just went to use / & it's giving me a 400 error
```
400 Bad Request
URISyntaxException thrown
```
I think it'll be easier for me to use a subdomain, so that I can use route path, rather than trying to fiddle with this. 
Thank you for your help.
